website/docs/guides/dbt-ecosystem/databricks-guides/how_to_optimize_dbt_models_on_databricks.md

```diff
@@ -0,0 +1,170 @@
+
+In addition, these tables are well suited for defining [dbt metrics](https://docs.getdbt.com/docs/build/metrics) on to ensure simplicity and consistency across your key business KPIs! Using the [dbt_metrics package](https://hub.getdbt.com/dbt-labs/metrics/latest/), you can query the metrics inside of your own dbt project even. With the upcoming Semantic Layer Integration, you can also then query the metrics in any of the partner integrated tools.
+
+### Filter rows in target and/or source
+
+It can be done using *incremental_predicates*
+
+```sql
+{{
+
+config(
+
+materialized='incremental',
+
+incremental_strategy = 'merge',
+
+unique_key = 'id',
+
+incremental_predicates = [
+
+"dbt_internal_target.create_at >= '2023-01-01'",	"dbt_internal_source.create_at >= '2023-01-01'"],
+
+)
+
+}}
+```
+
+## 3. Performance Troubleshooting
+
+Performance troubleshooting refers to the process of identifying and resolving issues that impact the performance of your dbt models and overall data pipelines. By improving the speed and performance of your Lakehouse platform, you will be able to process data faster, process large and complex queries more effectively, and provide faster time to market.  Let’s go into detail the three effective strategies that you can implement:
+
+### SQL warehouse query profile
+
+The SQL warehouse query profile is an effective tool found inside the Databricks SQL workspace. It’s used to troubleshoot slow-running queries, optimize query execution plans, and analyze granular metrics to see where compute resources are being spent. The query profile includes these high level capability areas:
+
+- Detailed information about the three main components of query execution, which are time spent in tasks, number of rows processed, and memory consumption.
+- Two types of graphical representations. A tree view to easily spot slow operations at a glance, and a graph view that breaks down how data is transformed across tasks.
+- Ability to understand mistakes and performance bottlenecks in queries.
+
+The three common examples of performance bottlenecks that can be surfaced by the query profile are:
+
+### Inefficient file pruning
+
+By default, Databricks Delta tables collect statistics on the **first 32 columns** defined in your table schema. When transforming data from the Bronze/staging layer to the Silver/intermediate layer, it is advised to reorder your columns to account for these file-level stats and improve overall performance. Move numerical keys and high cardinality query predicates to the left of the 32nd ordinal position, and move strings and complex data types after the 32nd ordinal position of the table. It is worth mentioning that while you can change the default table property to collect statistics on more columns, it will add more overhead as you write files. You may change this default value by using the [table property](https://docs.databricks.com/delta/table-properties.html), **delta.dataSkippingNumIndexedCols**.
```

Comment:
```suggestion
By default, Databricks Delta tables collect statistics on the _first 32 columns_ defined in your table schema. When transforming data from the Bronze/staging layer to the Silver/intermediate layer, it is advised to reorder your columns to account for these file-level stats and improve overall performance. Move numerical keys and high cardinality query predicates to the left of the 32nd ordinal position, and move strings and complex data types after the 32nd ordinal position of the table. It is worth mentioning that while you can change the default table property to collect statistics on more columns, it will add more overhead as you write files. You may change this default value by using the [table property](https://docs.databricks.com/delta/table-properties.html), `delta.dataSkippingNumIndexedCols`.
```